react/Image.tsx

```diff
@@ -114,20 +125,50 @@ function Image(props: ImageProps) {
    */
   const shouldOpenLinkInNewTab = link?.newTab ?? link?.openNewTab
 
+  const { push } = usePixel()
+
+  const promotionEventData =
+    analyticsProperties === 'provide'
+      ? {
+          id: promotionId,
+          name: promotionName,
+          creative: formattedSrc,
+          position: promotionPosition,
+        }
+      : undefined
+
   const maybeLink = link ? (
     <a
       href={formatIOMessage({ id: link.url, intl })}
       rel={link.noFollow ? 'nofollow' : ''}
       target={shouldOpenLinkInNewTab ? '_blank' : undefined}
       title={formatIOMessage({ id: link?.attributeTitle, intl })}
       className={handles.imageElementLink}
+      onClick={() => {
+        if (analyticsProperties === 'none') return
+
+        push({ event: 'promotionClick', promotions: [promotionEventData] })
+      }}
     >
       {imgElement}
     </a>
   ) : (
     <Fragment>{imgElement}</Fragment>
   )
 
+  useOnView({
+    ref: imageRef,
+    onView: () => {
+      if (analyticsProperties === 'none') return
```

Comment:
the if here can be placed on the outside to not call the useOnView when analyticsProperties is `none`?

Reply:
We can't call hooks conditionally 😢